### PR TITLE
Docker image for 1.4.12

### LINKED ISSUE #336: Unable to connect to Mosquitto 1.4.10 over Websocket
Hello,

I am unable to connect to my local Mosquitto 1.4.10 broker from a JavaScript client over a websocket.

The same JavaScript client is successfully connecting to the public broker at test.mosquitto.org on port 8080 over a websocket.

The MQTT protocol connection on port 1883 is working fine, which I tested using mosquitto_pub and mosquitto_sub.

My broker is set up within a VirtualBox running Ubuntu 14.04. 

I have libwebsockets installed on the same virtual machine.

My local broker was compiled with **WITH_WEBSOCKETS:=yes** in the _config.mk_ file

I confirmed by running _sudo netstat_ that the Websockets port is open.

I am loading the JavaScript client web page from a Firefox browser, from inside the same virtual machine on which Mosquitto is running. I get the following error message in the browser console (Apache web server is running on the same virtual machine as Mosquitto):

> 	Firefox can't establish a connection to the server at ws://localhost:8080/mqtt

I did a trace through Wireshark and it seems that the client is getting no response from the server after sending the initial HTTP request to upgrade to the Websockets protocol (see trace below).

As far as I can tell, the issue lies in the Mosquitto installation. Whether it is a bug or something missing in my setup, I cannot tell.

It might help, for starters, if I could look at the .conf file set up for the test.mosquitto.org installation, so I may compare it to my settings.

Any other suggestions on fixing this will be greatly appreciated.

Thanks.

Here is my Mosquitto .conf file:

```
 	port 1883
 	listener 8080
 	protocol websockets
 
 	log_type all
 	websockets_log_level 1023
 	connection_messages true
```

Here is the Mosquitto server's log (with websockets logging level set to 1023, and verbose logging turned on - no messages appear when I load the JavaScript web page):

> 1481381105: mosquitto version 1.4.10 (build date 2016-12-10 18:47:37+0530) starting
> 1481381105: Config loaded from /etc/mosquitto/mosquitto.conf.
> 1481381105: Opening websockets listen socket on port 8080.
> 1481381105: Initial logging level 1023
> 1481381105: Libwebsockets version: 2.1.0 manavkumarm@manav-alljoyn-
> 1481381105: IPV6 not compiled in
> 1481381105: libev support not compiled in
> 1481381105: libuv support not compiled in
> 1481381105:  Threads: 1 each 1024 fds
> 1481381105:  mem: platform fd map:  4096 bytes
> 1481381105:  Compiled with OpenSSL support
> 1481381105: Creating Vhost 'default' port 8080, 3 protocols, IPv6 off
> 1481381105:  Using non-SSL mode
> 1481381105:  Listening on port 8080
> 1481381105:  mem: per-conn:          376 bytes + protocol rx buf
> 1481381105:  canonical_hostname = mqtt
> 1481381105: Opening ipv4 listen socket on port 1883.
> 1481381105: Opening ipv6 listen socket on port 1883.

Here is the Wireshark trace:

> GET /mqtt HTTP/1.1
> Host: localhost:8080
> User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux i686; rv:49.0) Gecko/20100101 Firefox/49.0
> Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
> Accept-Language: en-US,en;q=0.5
> Accept-Encoding: gzip, deflate
> Sec-WebSocket-Version: 13
> origin: http://localhost
> Sec-WebSocket-Protocol: mqtt
> Sec-WebSocket-Extensions: permessage-deflate
> Sec-WebSocket-Key: 2jGKB8w/B3al/XajJp3o6g==
> Connection: keep-alive, Upgrade
> Pragma: no-cache
> Cache-Control: no-cache
> Upgrade: websocket

### PR DESCRIPTION
**Pull request by onip**
added Dockerfile and other stuff for building docker image based on version 1.4.12.

In my tests this also fixes #336

### CODE REVIEW

**Comment by karlp** [comment]
Is there any reason to keep the older 1.4.x docker configs?

File: docker/1.4.12/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/mosquitto/log
+```
+
+
+##Configuration
+
+When running the image, the default configuration values are used.
+To use a custom configuration file, mount a **local** configuration file to `/mosquitto/config/mosquitto.conf`
+```
+docker run -it -p 1883:1883 -p 9001:9001 -v <path-to-configuration-file>:/mosquitto/config/mosquitto.conf mosquitto:1.4.10
```
Comment (karlp) [inline docker/1.4.12/README.md]: This is 1.4.12 not 1.4.10 right?

File: docker/1.4.12/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+log_dest file /mosquitto/log/mosquitto.log
+```
+
+**Note**: If a volume is used, the data will persist between containers.
+
+##Build
+Build the image:
+```
+docker build -t mosquitto:1.4.10 .
```
Comment (karlp) [inline docker/1.4.12/README.md]: again, wrong version.

File: docker/1.4.12/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+##Run
+Run a container using the new image:
+```
+docker run -it -p 1883:1883 -p 9001:9001 -v <path-to-configuration-file>:/mosquitto/config/mosquitto.conf -v /mosquitto/data -v /mosquitto/log mosquitto:1.4.10
```
Comment (karlp) [inline docker/1.4.12/README.md]: and here.

Comment (onip) [inline docker/1.4.12/README.md]: I was in a hurry and completely missed those. Should be fixed now.

Also I've updated docker README to use the official image name from docker hub.

**Comment by ralight** [comment]
Thanks very much for your contribution!

**Pull request merged**